docs/configuration.md

```diff
@@ -78,6 +78,98 @@ TOML is also supported in `.release-it.toml`:
 "before:init" = "npm test"
 ```
 
+## Configuration options
+
+Release-it has [plenty of options][2] to configure. See the table below for
+descriptions of each.
+
+### Hooks
+
+| Option | Description |
+| --- | --- |
+| `hooks` | Use script hooks to run shell commands at any moment during the release process (such as `before:init` or `after:release`). |
+
+### Git
+
+| Option | Description |
+| --- | --- |
+| `git.changelog` | Override the changelog generation behavior, which is based on `git log` by default. |
+| `git.requireCleanWorkingDir` | Require that all file changes are committed. |
+| `git.requireBranch` | Require that the release is on a particular branch name (e.g. `dev`), or `false` to disable this option. |
+| `git.requireUpstream` | Require that an upstream remote exists. Disable this option to release from a branch that has not yet been published to a remote, such as from a brand new project. |
+| `git.requireCommits` | Stop the process if there are no commits since the previous release. |
+| `git.requireCommitsFail` | If there are no commits, halt the release with exit code 0, but don't stop the process. This may be used in CI so that the whole pipeline doesn't fail. |
+| `git.commitsPath` | The path to the directory that should be included in the release changes. |
+| `git.addUntrackedFiles` | If `true`, add untracked files to the release commit. |
+| `git.commit` | If `false`, skip the commit release step. |
+| `git.commitMessage` | The message to add to the commit step. |
+| `git.commitArgs` | In case extra arguments should be provided to Git for commit args. |
+| `git.tag` | If `false`, skip the tag release step |
+| `git.tagExclude` | String used to override the normal behavior to find the latest tag. I.e. to exclude pre-releases from a tag, set this to `*[-]*` to exclude everything with a hyphen, which is normally used exclusively in pre-releases. Note that `git.tagExclude` has no effect when `git.getLatestTagFromAllRefs: true` is set. |
+| `git.tagName` | String used to set a custom tag name, which may not be the same as the (prefixed) version. |
+| `git.tagMatch` | Glob string used to override the normal matching behavior to find the latest tag. Note that this uses a glob, not a regex. |
+| `git.getLatestTagFromAllRefs` | If `true`, considers all tags (sorting them by version), whether directly reachable or not (e.g. include tags that point to sibling commits on different branches) |
+| `git.tagAnnotation` | _TODO: Docs to be added._ |
```

Comment:
This looks to me like it is used for the tag message.  Not sure why this is called `tagAnnotation` instead of `tagMessage`.

Reply:
Added in https://github.com/release-it/release-it/pull/1155/commits/e3c93ed9bb2a75bebb93be57eaad237d9fa9f23a